### ISSUE TITLE
[Ray AIR] 

### ISSUE THREAD

**Issue by dmatrix** [ray-project/ray#29640]
### What happened + What you expected to happen

The training team and I encountered this problem trying to use Ray AIR to do an e2e example.

Data: NYC Tax Cab
Trainer: XGBoosTrainer
Code: [Notebook](https://github.com/dmatrix/misc-code/blob/master/py/ray/rayair/examples/nyc-taxi/e2e_nyc.ipynb)

Run this notebook as is and you will get the error. 

Commet out `valid_dataset = valid_dataset.drop_columns(["is_big_tip"])` line and exclude valid dataset from the `Train` will run successfully.

This is blocking us from gathering validity metrics during training since we cannot include the validation dataset in the Training

The workaround is to remove validation data, which is a problem, because we then can't compute and track validation loss.

### Versions / Dependencies

Ray 2.0
Python 3.8.1

### Reproduction script

Code: [Notebook](https://github.com/dmatrix/misc-code/blob/master/py/ray/rayair/examples/nyc-taxi/e2e_nyc.ipynb)

### Issue Severity

High: It blocks me from completing my task. cc @emmyscode

**Comment by amogkam** [comment]
@dmatrix can you share the stack trace you are seeing if you keep the `drop_columns` line?

**Comment by dmatrix** [comment]
![Screen Shot 2022-10-24 at 6 09 33 PM](https://user-images.githubusercontent.com/1117597/197660059-4908452d-ed4c-4aa0-9502-182a5879c339.png)
![Screen Shot 2022-10-24 at 6 10 01 PM](https://user-images.githubusercontent.com/1117597/197660064-f87271e5-b79b-423a-97e1-df374239d56f.png)

**Comment by emmyscode** [comment]
Adding on that if you **keep** the validation data, but do **not** drop the target column "is_big_tip", the code runs successfully. Is this the intended behavior?

**Comment by amogkam** [comment]
Discussed with @dmatrix right now- for validation data you want to keep the label column. When doing validation you still calculate the loss and accuracy, requiring the label column.

However, for a _test_ dataset, you would drop the label column. But then for this case, you would not pass this into `XGBoostTrainer`, but instead use it with `BatchPredictor` after training has finished.

**Comment by amogkam** [comment]
https://stats.stackexchange.com/questions/19048/what-is-the-difference-between-test-set-and-validation-set

**Comment by amogkam** [comment]
That being said, I think we can raise a better error message here on the xgboost side.

**Comment by dmatrix** [comment]
Agreed. The error was a bit misleading.
